spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDefinitionController.java

```diff
@@ -141,6 +141,9 @@ public void save(@RequestParam("name") String name,
 					@RequestParam(value = "deploy", defaultValue = "false")
 					boolean deploy) {
 		StreamDefinition stream = new StreamDefinition(name, dsl);
+		if (this.repository.findOne(stream.getName()) != null) {
```

Comment:
I think this logic should move down into the repository implementation, as that is what is done for the current in-memory implementation.  Also, the save(Iterable) method in the repo should delegate to the save() method so that the check for existence can be done on each of the definitions.